feature-libs/asm/components/customer-selection/customer-selection.component.ts

```diff
@@ -117,10 +124,92 @@ export class CustomerSelectionComponent implements OnInit, OnDestroy {
 
   closeResults() {
     this.asmService.customerSearchReset();
+    this.setFocusOnInput();
   }
 
   ngOnDestroy(): void {
     this.subscription.unsubscribe();
     this.asmService.customerSearchReset();
   }
+
+  /**
+   * set focus to the first searched item
+   * @param event keyboard event
+   */
+  focusFirstItem(event: UIEvent): void {
+    event.preventDefault();
+    this.activeIndex = 0;
+    this.updateItemIndex(this.activeIndex);
+  }
+  /**
+   * set mouse cursor to the end of search text
+   * @param event keyboard event
+   */
+  setSelectionEnd(event: UIEvent): void {
+    event.preventDefault();
+    if (this.searchTerm.nativeElement.value?.length) {
+      const selectionStart = this.searchTerm.nativeElement.value.length;
+      this.searchTerm.nativeElement.selectionStart = selectionStart;
+      this.searchTerm.nativeElement.selectionEnd = selectionStart;
+    }
+  }
+  /**
+   * set focus on previous searh result item.  If no previous item then go to end of item.
+   * @param event keyboard event
+   */
+  focusPreviousChild(event: UIEvent): void {
+    event.preventDefault();
+    this.activeIndex--;
+    if (this.activeIndex < 0) {
+      this.activeIndex = this.resultItems.length - 1;
+    }
+    this.updateItemIndex(this.activeIndex);
+  }
+  /**
+   * set focus on next searh result item.  if no next item then go to the first item
+   * @param event keyboard event
+   */
+  focusNextChild(event: UIEvent): void {
+    event.preventDefault();
+    this.activeIndex++;
+    if (this.activeIndex > this.resultItems.length - 1) {
+      this.activeIndex = 0;
+    }
+    this.updateItemIndex(this.activeIndex);
+  }
+  /**
+   * set focus to input search text
+   * @param event keyboard event
+   * @param {boolean} moveOneLeft - flag to move cursor one left
+   * @param {boolean} moveBegin - flag to move cursor to front
+   */
+  focusInputText(event: UIEvent, moveOneLeft = false, moveBegin = false): void {
```

Comment:
Can these properties be specified together?

Reply:
no and we will change to enum for these last parameters ~~~

Reply:
I will remove last two parameters because from the event object i could get which key has pressed